### ISSUE TITLE
[HOLD for payment 2025-01-28] [#expense] [Bring copilot to NewDot] Re-add the Try New Expensify button (when applicable to show it) within HybridApp for Copilot

### ISSUE THREAD

**Issue by dangrous** [Expensify/App#51264]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

E/A issue for https://github.com/Expensify/Expensify/issues/414571 . We removed this functionality before copilot was migrated over, and now it is! @war-in is working on PRs already.

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight</details>

**Comment by war-in** [comment]
Hi 👋 I'm working on PRs here ([ND part](https://github.com/Expensify/App/pull/50800), [OD part](https://github.com/Expensify/Mobile-Expensify/pull/13220))

**Comment by dangrous** [comment]
assigned!

**Comment by dangrous** [comment]
@war-in if they're ready, can you put the PRs into review? You can add a [HOLD] to the title so they won't be merged early

**Comment by war-in** [comment]
> @war-in if they're ready, can you put the PRs into review? You can add a [HOLD] to the title so they won't be merged early

PRs are marked as ready for review :) Mentioned you in both 🫡

**Comment by dangrous** [comment]
@Ollyws you were assigned to review the App PR, let us know if you can take a look soon, thanks!

**Comment by war-in** [comment]
Hi 👋 I updated both PRs with the newest main and I'm waiting for the final review

**Comment by Julesssss** [comment]
This can't be tested by External contributors, so I'll unassign you @Ollyws. 

Also, @dangrous I can take over reviewing and testing of the PRs

**Comment by Julesssss** [comment]
Alright, next step is to wait for the App changes to hit prod then I'll bump the HybridApp submodule ([like this](https://github.com/Expensify/Mobile-Expensify/pull/13272)) and test @war-in's HybridApp changes.

Back in a few days.

**Comment by dangrous** [comment]
Thanks @Julesssss !

**Comment by dangrous** [comment]
@war-in do you want to give a summary of that other bug with the tokens you were encountering here, so @Julesssss can weigh in? So for some reason disconnecting as a delegate calls CreateLogin which then fails [here](https://github.com/Expensify/Auth/blob/4bb1d5e6a408d5bcad57f389e73cde8425dc49dd/auth/command/CreateLogin.cpp#L55-L63) because it's sending [`chat-expensify-com`](https://github.com/Expensify/Auth/blob/9e48ecb7f74857e35bcd0e65b6260c27150ee37d/auth/lib/Partner.cpp#L36) as the partner name. I think we could probably just add that partner name to the conditional there to fix it, but I'm not sure if that would cause other side effects. I'm happy to keep working on that side of things, but would love your input!

**Comment by Julesssss** [comment]
> So for some reason disconnecting as a delegate calls CreateLogin which then fails [here](https://github.com/Expensify/Auth/blob/4bb1d5e6a408d5bcad57f389e73cde8425dc49dd/auth/command/CreateLogin.cpp#L55-L63) because it's sending [chat-expensify-com](https://github.com/Expensify/Auth/blob/9e48ecb7f74857e35bcd0e65b6260c27150ee37d/auth/lib/Partner.cpp#L36) as the partner name.

Oh thanks for sharing, was that discussion on another issue or something? Adding `chat-expensify-com` as a partner seems reasonable to me, maybe we should just ask in slack to see if anyone see's a problem with it. But yeah it would be great if you can continue with that and I'll handle the HybridApp reviews and tests 👍

**Comment by war-in** [comment]
> Oh thanks for sharing, was that discussion on another issue or something?

I DMed @dangrous to check if it's a known issue or not and forgot to write a summary here, sorry!

**Comment by war-in** [comment]
FYI, there's also the second [PR](https://github.com/Expensify/Mobile-Expensify/pull/13220) (on the OD side) so let's not close the issue yet

**Comment by dangrous** [comment]
Not overdue, working on existing PRs, and I'll hopefully be pushing up 1-2 PRs today to fix the token issue for Hybrid App

**Comment by Julesssss** [comment]
Multiple PRs ongoing.

**Comment by dangrous** [comment]
Not overdue, the PRs that allow this to move forward are [in review](https://github.com/Expensify/App/pull/52980) and then [waiting](https://github.com/Expensify/Web-Expensify/pull/44547) on the one in review. The App PR is good to go with current backend, and now being tested with the changes. We'll be there soon!

**Comment by dangrous** [comment]
Working on one minor fix in the backend PR and then can move forward.

**Comment by dangrous** [comment]
nearly there!

**Comment by dangrous** [comment]
App is deployed - the backend PR is not yet merged but I **think** we won't need it to move forward with testing this and getting it done.

**Comment by dangrous** [comment]
both parts merged! So we're just waiting on https://github.com/Expensify/Mobile-Expensify/pull/13289 again

**Comment by dangrous** [comment]
We're almost done here but are shooting for doing this as soon as 2025 rolls around since both @war-in and I are out through new years and we want to be around to deal with anything unexpected.

**Comment by Julesssss** [comment]
Sounds like [the PR](https://github.com/Expensify/App/issues/51264#issuecomment-2557746541) can soon be finished up.

**Comment by dangrous** [comment]
Nearly here! Will need one backend change, and then final review of the two associated PRs - https://github.com/Expensify/App/pull/54296 and https://github.com/Expensify/Mobile-Expensify/pull/13289. Then we're good to go!

**Comment by dangrous** [comment]
Okay we're back on track, estimated merge is Friday

**Comment by dangrous** [comment]
50% there - ND PR is merged, Mobile-Expensify is in final review from SWM
